webui/react/src/components/TaskActionDropdown.tsx

```diff
@@ -19,39 +21,73 @@ interface Props {
   curUser?: DetailedUser;
   onComplete?: (action?: Action) => void;
   onVisibleChange?: (visible: boolean) => void;
-  task: AnyTask;
+  task: CommandTask;
 }
 
 const TaskActionDropdown: React.FC<Props> = ({ task, onComplete, children }: Props) => {
   const { canModifyWorkspaceNSC } = usePermissions();
-  const isKillable = isTaskKillable(
-    task,
-    canModifyWorkspaceNSC({ workspace: { id: task.workspaceId } }),
-  );
+
+  const isConnectable = (task: CommandTask) => {
+    const connectableTaskTypes: CommandType[] = [CommandType.JupyterLab, CommandType.Shell];
+    return connectableTaskTypes.includes(task.type) && task.state === CommandState.Running;
+  };
 
   const confirm = useConfirm();
 
-  const menuItems: MenuItem[] = [
-    {
-      key: Action.ViewLogs,
-      label: 'View Logs',
-    },
-  ];
+  const taskConnectFields = useMemo(() => {
+    switch (task.type) {
+      case CommandType.JupyterLab:
+        return [
+          {
+            label: 'Connect to the running Jupyter Server in VSCode:',
+            value: `${serverAddress()}${task.serviceAddress}`,
+          },
+        ];
+      case CommandType.Shell:
+        return [
+          {
+            label: 'Start an interactive SSH session in the terminal:',
+            value: `det shell open ${task.id}`,
+          },
+        ];
+      default:
+        return [];
+    }
+  }, [task]);
 
-  if (isKillable) menuItems.unshift({ key: Action.Kill, label: 'Kill' });
+  const TaskConnectModal = useModal(TaskConnectModalComponent);
```

Comment:
nit: by convention, this should be declared on top of the component. around `const { canModifyWorkspaceNSC } = usePermissions();`